CHANGELOG.md

```diff
@@ -149,6 +149,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Moved `block_ddp_sync_behaviour` out of `TrainingBatchLoop` to loop utilities ([#9192](https://github.com/PyTorchLightning/pytorch-lightning/pull/9192))
 
 
+- Avoid wrapping LightningModule with data-parallel modules when not fitting in `DDPPlugin`, `DDPSpawnPlugin`, `DDPShardedPlugin`, `DDPSpawnShardedPlugin` ([#6977]https://github.com/PyTorchLightning/pytorch-lightning/issues/6977)
+
+
```

Comment:
i think this is a bugfix. before, if someone had no trainable parameters in their model and they called trainer.test(), Lightning would raise an exception because DDP construction would raise an exception around no parameters with requires_grad found. 

this error is compounded by the process reconciliation check. when `configure_ddp` raises an exception, we don't call `_share_information_to_prevent_deadlock`

as a result, `sync_dir` is not initialized. 

which causes this to fail: https://github.com/PyTorchLightning/pytorch-lightning/blob/35876bb75f27eb8f44220afd4bfa757a0432d233/pytorch_lightning/plugins/training_type/ddp.py#L481-L486

because `sync_dir` is None and we're not checking it 😞